FrostCord/FrostCord
# Direct Messaging

# Direct Messaging
This PR introduces a DM system for you to message users privately in a 1-on-1 chat

# API Changes
- Adds `directmessage.service.ts` which provides services to create/fetch DM channels
- Completes the form in mini profiles allowing you to send a DM from a user's mini profile
- Implements a context menu to open a DM channel with a user (or if one exists, navigate to that one)
- Adds the `Directs` sidebar to the Friends tab
- Removes the message form for the user's own mini profile
- Removes the `messages` tab from the navbar

# Testing
- Verify the context menus when right clicking a user's message header in chat allows you to DM them, and that only one DM channel between you and the recipient can exist (when using it again, you should be navigated to the same DM channel that was made on first use of it)
- Verify the same applies for the message button in the friends list
- Verify you can DM a user from a mini profile by typing in the message bar and hitting enter to send. The same rules from above should apply and you should be navigated to the DM with your message sent
- Verify you cannot see DM channels made by other users that are NOT addressed to you
- Verify DM servers aren't populated in the main serverlist
- Verify the system behaves consistently
